source/retryable-reads/tests/README.rst

```diff
@@ -164,6 +165,45 @@ Optional Enumeration Commands
 A driver only needs to test the optional enumeration commands it has chosen to
 implement (e.g. ``Database.listCollectionNames()``).
 
+PoolClearedError Retryability Test
+==================================
+
+This test will be used to ensure drivers properly retry after encountering PoolClearedErrors.
+This test MUST be implemented by any driver that implements the CMAP specification.
+
+1. Create a client with maxPoolSize=1 and retryReads=true. If testing against a
+   sharded deployment, be sure to connect to only a single mongos.
+
+2. Enable the following failpoint::
+
+     {
+         configureFailPoint: "failCommand",
+         mode: { times: 1 },
+         data: {
+             failCommands: ["find"],
+             errorCode: 91,
+             blockConnection: true,
+             blockTimeMS: 1000
+         }
+     }
+
+3. Start two threads and attempt to perform a ``findOne`` simultaneously on both.
+
+4. Verify that both ``findOne`` attempts succeed.
+
+5. Via CMAP monitoring, assert that the first check out succeeds.
+
+6. Via CMAP monitoring, assert that a PoolClearedEvent is then emitted.
+
+7. Via CMAP monitoring, assert that the second check out then fails due to a
+   connection error.
```

Comment:
The CMAP spec doesn't require that drivers include any information about the actual error that occurred, should we update it to do so? Could drivers do that without a breaking change? cc @divjotarora